reposilite-plugins/prometheus-plugin/src/main/kotlin/com/reposilite/plugin/prometheus/PrometheusPlugin.kt

```diff
@@ -32,38 +41,77 @@ class PrometheusPlugin : ReposilitePlugin() {
             ?: throw IllegalStateException("Prometheus password is not defined")
 
         val prometheusFacade = PrometheusFacade(
-            failureFacade = facade(),
+            failureFacade = facade<FailureFacade>(),
             prometheusUser = prometheusUser,
             prometheusPassword = prometheusPassword
         )
 
+        JvmMetrics.builder().register()
+        logger.info("Prometheus | JVM metrics has been initialized.")
+
+        val reposiliteMetrics = ReposiliteMetrics.builder {
+            statisticsFacade = facade<StatisticsFacade>()
+            statusFacade = facade<StatusFacade>()
+            failureFacade = facade<FailureFacade>()
+        }.register()
+
+        event { event: ResolvedFileEvent ->
+            event.result.map { (info, _) ->
+                reposiliteMetrics.responseFileSizeSummary.observe(info.contentLength.toDouble())
+                reposiliteMetrics.resolvedFileCounter.inc()
+            }
+        }
+
+        event { _: DeployEvent ->
+            reposiliteMetrics.mavenDeployCounter.inc()
+        }
+
+        logger.info("Prometheus | Reposilite metrics has been initialized")
+
+
         event { event: HttpServerConfigurationEvent ->
             val server = event.config.pvt.jetty.server!!
+            val handler = StatisticsHandler()
+            server.handler = handler
+
+            val jettyMetrics = JettyStatisticsHandlerMetrics.builder {
+                statisticsHandler = handler
+            }.register()
+
+            // Abusing request log, as it's the only way to get the time the request was sent
+            server.setRequestLog { request, response ->
+                jettyMetrics.responseTimeSummary.labelValues(response.status.toString())
+                    .observe((System.currentTimeMillis() - request.timeStamp).milliseconds.toDouble(DurationUnit.SECONDS))
```

Comment:
We could probably just get current time in the Javalin's before handler and store it in the ctx.attribute, so later on we can get it back in the after handler. Might be a good idea to add it as a new event to the core impl :thinking:

Reply:
yeah that would be possible as well

tbh the request does probably have the time, it just needs to be cast to `Request` (yucky)

Reply:
I don't remember, you'd need to jump in with the debugger to investigate all of these underlying properties. Probably just faster to do it on our side for now :sweat_smile: